CHANGELOG.md

```diff
@@ -9,9 +9,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 - Add `include_deleted` to ledger_entry request
+- Add support for the `PermissionedDomains` feature
 
 ### BREAKING CHANGE:
 - Remove Python 3.7 support to fix dependency installation and use 3.8 as new default.
+- Remove the AccountObjectType enum. Instead, use the LedgerEntryType enum, this reduces redundancy in the codebase.
```

Comment:
Not all valid `LedgerEntryType`s are valid `AccountObjectType`s.

Reply:
Not to mention, it's also confusing.

Reply:
the xrpl.js library does not have separate filters for LedgerEntryType and AccountObjectType

Reply:
I think its a good opportunity to refactor and clean up the code. Its slow to wait for approvals of multiple small PRs

Reply:
> the xrpl.js library does not have separate filters for LedgerEntryType and AccountObjectType

Yes it does: https://github.com/XRPLF/xrpl.js/blob/24e9ad7c12582f66e6391a1aa362a3e760d06dbf/packages/xrpl/src/models/methods/accountObjects.ts#L6-L9

It may depend on the `LedgerEntryType` equivalent, but it's still separate.

Reply:
I agree that some ledger-entries are excluded from AccountObjectType. In fact, the xrpl.js implementation is faulty. `rippled` does not allow five `ledger_entry_type`'s in the `account_objects` RPC: https://github.com/XRPLF/rippled/blob/0ec17b6026298dc150099b66a2cecad0fe561d1b/src/xrpld/rpc/detail/RPCHelpers.cpp#L994

I didn't narrow the type because `rippled` will raise an error if incompatible types are used in the filter.

Having said that, I still believe its useful to unify the `AccountObjectType` and `LedgerEntryType` in xrpl-py codebase.

Reply:
If you feel strongly that this refactor shouldn't be mixed in with PD amendment, I can roll the commit back